api/v1beta1/charmedk8sconfig_types.go

```diff
@@ -28,8 +28,10 @@ type CharmedK8sConfigSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of CharmedK8sConfig. Edit charmedk8sconfig_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// List of Juju applications to deploy to control plane machines
+	ControlPlaneApplications []string `json:"controlPlaneApplications,omitempty"`
```

Comment:
Should these be units instead of applications?

Reply:
I don't think so; the intention is for the Juju provider to call [AddUnits](https://github.com/juju/juju/blob/7f79e8a1bbf0877b93e820ee84a00e6caffc53f5/api/client/application/client.go#L367-L392) to create new units for the listed applications, placed on the correct machine. The units don't exist prior to that, so this is a list of applications for which new units should be created.

Reply:
ok, I see now. Thanks!